javaagent-muzzle-generation/build.gradle.kts

```diff
@@ -0,0 +1,22 @@
+plugins {
+  `kotlin-dsl`
+
+  id("otel.java-conventions")
+  id("otel.publish-conventions")
+  id("com.gradle.plugin-publish")
+}
+
+group = "io.opentelemetry.instrumentation.gradle"
+
+dependencies {
+  implementation("com.google.guava:guava")
+  implementation(project(":muzzle"))
+  implementation("net.bytebuddy:byte-buddy-gradle-plugin")
+  implementation("io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api")
+}
+
+pluginBundle {
+  website = "https://opentelemetry.io"
+  vcsUrl = "https://github.com/open-telemetry/opentelemetry-java-instrumentation"
+  tags = listOf("opentelemetry", "instrumentation")
+}
```

Comment:
We'll need that `plugins` snippet from my draft PR when publishing to portal so may as well add it now right?

Reply:
Which PR? :) Sorry, you are too laconic for me recently :(

Reply:
Sorry for my smartphone replies, somehow GitHub app still doesn't support linking to lines of code. And now that I look for it I can't find it in the PR I sent, that's definitely too laconic...

```
gradlePlugin {
    plugins { 
        named("io.opentelemetry.instrumentation.javaagent-codegen") { 
            displayName = "<short displayable name for plugin>" 
            description = "<Good human-readable description of what your plugin is about>" 
        }
    }
}
```

https://docs.gradle.org/current/userguide/publishing_gradle_plugins.html#configure_the_plugin_publishing_plugin

One thing I found is for precompiled script plugin, we leave out `id` and `implementationClass` and it seems to work ok